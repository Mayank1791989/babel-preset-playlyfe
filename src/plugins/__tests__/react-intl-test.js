/* @flow */
import { transformFileSync, transform } from 'babel-core';
import path from 'path';

const fixtureFile = path.join(__dirname, './data', 'component.js');

describe('check intl messages in transformed file metadata', () => {
  const { metadata } = transformFileSync(fixtureFile, {
    babelrc: false,
    presets: [require('babel-preset-env')],
    plugins: [
      require('babel-plugin-transform-object-rest-spread'),
      require('babel-plugin-syntax-jsx'),
      require('babel-plugin-transform-react-jsx'),
      require('../react-intl').default,
    ],
  });
  it('exports 6 intl messages', () => {
    expect(metadata['react-intl'].messages).toMatchSnapshot();
  });
});

test('correctly throw invalid intl message error', () => {
  expect(() => {
    transform(`
      import React from 'react';
      import { FormattedMessage } from 'react-intl';

      const Component = (
        <FormattedMessage
          id="test.xyz"
          defaultMessage="{xyz, select, y {test}"
        />
      );
    `, {
      babelrc: false,
      presets: [require('babel-preset-env')],
      plugins: [
        require('babel-plugin-transform-object-rest-spread'),
        require('babel-plugin-syntax-jsx'),
        require('babel-plugin-transform-react-jsx'),
        require('../react-intl').default,
      ],
    });
  }).toThrowErrorMatchingSnapshot();
});
