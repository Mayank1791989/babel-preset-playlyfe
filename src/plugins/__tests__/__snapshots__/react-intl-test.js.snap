// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check intl messages in transformed file metadata exports 6 intl messages 1`] = `
Array [
  Object {
    "defaultMessage": "External Define Messasssge",
    "description": undefined,
    "id": "externaslDefineMessage",
  },
  Object {
    "defaultMessage": "Inline Text 1",
    "description": undefined,
    "id": "inlineComponent",
  },
  Object {
    "defaultMessage": "Inline Text 1",
    "description": undefined,
    "id": "inlineHTMLComponent",
  },
  Object {
    "defaultMessage": "Inline DefineMessage",
    "description": undefined,
    "id": "inlineDefineMessage",
  },
  Object {
    "defaultMessage": "Inline Format Message",
    "description": undefined,
    "id": "inlineFormatMessage",
  },
  Object {
    "defaultMessage": "Inline Format HTML Message",
    "description": undefined,
    "id": "inlineFormatHTMLMessage",
  },
]
`;

exports[`correctly throw invalid intl message error 1`] = `
"undefined: [React Intl] Message failed to parse. See: http://formatjs.io/guides/message-syntax/
SyntaxError: Expected \\"=\\", \\"\\\\\\\\#\\", \\"\\\\\\\\\\\\\\\\\\", \\"\\\\\\\\u\\", \\"\\\\\\\\{\\", \\"\\\\\\\\}\\", \\"}\\" or [^{}\\\\\\\\\\\\0-\\\\x1F\\\\x7f \\\\t\\\\n\\\\r] but end of input found.
[0m [90m  6 | [39m        [33m<[39m[33mFormattedMessage[39m[0m
[0m [90m  7 | [39m          id[33m=[39m[32m\\"test.xyz\\"[39m[0m
[0m[31m[1m>[22m[39m[90m  8 | [39m          defaultMessage[33m=[39m[32m\\"{xyz, select, y {test}\\"[39m[0m
[0m [90m    | [39m                         [31m[1m^[22m[39m[0m
[0m [90m  9 | [39m        [33m/[39m[33m>[39m[0m
[0m [90m 10 | [39m      )[33m;[39m[0m
[0m [90m 11 | [39m    [0m"
`;
